main.py

```diff
@@ -0,0 +1,59 @@
+import sys
+
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+import requests
+from bs4 import BeautifulSoup
+from googletrans import Translator
+def check_url(url):
+    r = requests.head(url)
+    return r.status_code == 200
+
+def get_text_from_site(url):
+    try:
+        response = requests.get(url)
+        bs = BeautifulSoup(response.text, 'html.parser')
+        part_of_text = bs.find('div', 'c-detail text').text
+        index = part_of_text.lower().find("приметы")
+        i = part_of_text.find(":", index)
+        result_sentence = part_of_text[i+1:part_of_text.find("\n", i+1)].replace(".", "")
+        translator = Translator()
+        translation = translator.translate(result_sentence, src='ru', dest='en')
```

Comment:
`googletrans` is a very poor translator. If you want to use external API better to use [`deep-translator`](https://github.com/nidhaloff/deep-translator). You can select Google API (but be careful with RPS, no more than 2-3 requests per second) or Yandex/DeepL which can be better for ru-en pair.

But you can also use another Neural Network from HF, e.g. https://huggingface.co/facebook/nllb-200-distilled-600M. Or the same NLLB, but w/ 600M parameters (it's bigger, but better)